operators/constellation-node-operator/controllers/joiningnode_controller.go

```diff
@@ -73,23 +80,35 @@ func (r *JoiningNodesReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return r.Update(ctx, &node)
 	})
 	if err != nil {
-		logr.Error(err, "unable to update Node")
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		// check if the deadline has been reached
+		// requeue if not
+		if joiningNode.Spec.Deadline == nil || r.Now().Before(joiningNode.Spec.Deadline.Time) {
```

Comment:
`r.Now().Before(joiningNode.Spec.Deadline.Time)` doesn't this evaluate to `true` in case the deadline is still in the future?

Reply:
Yes. If the deadline is in the future, we re-queue the request to have a look at it again.